src/tests/src/modeler/loadFiles/test-load-optim-config.cpp

```diff
@@ -0,0 +1,300 @@
+#define WIN32_LEAN_AND_MEAN
+
+#include <filesystem>
+#include <fstream>
+#include <unit_test_utils.h>
+
+#include <boost/test/unit_test.hpp>
+
+#include <antares/solver/modeler/loadFiles/loadFiles.h>
+#include "antares/study/system-model/optimConfig.h"
+
+using namespace Antares::Solver::LoadFiles;
+using namespace Antares::Modeler::Config;
+
+struct CreateInputFileFixture
+{
+    CreateInputFileFixture();
+    void createLibraryFile(const std::string& yaml_content);
+    void createOptimConfigFile(const std::string& yaml_content);
+    ~CreateInputFileFixture();
+
+    std::filesystem::path studyFolder;
+
+private:
+    void createYamlFile(const std::string& filename, const std::string& yaml_content);
+    std::filesystem::path inputFolder_;
+    std::filesystem::path librariesFolder_;
+    std::filesystem::path creationFolder_;
+};
+
+CreateInputFileFixture::CreateInputFileFixture()
+{
+    studyFolder = std::filesystem::temp_directory_path();
+
+    inputFolder_ = studyFolder / "input";
+    std::filesystem::create_directory(inputFolder_);
+
+    librariesFolder_ = inputFolder_ / "model-libraries";
+    std::filesystem::create_directory(librariesFolder_);
+}
+
+void CreateInputFileFixture::createLibraryFile(const std::string& yaml_content)
+{
+    creationFolder_ = librariesFolder_;
+    createYamlFile("my-library.yml", yaml_content);
+}
+
+void CreateInputFileFixture::createOptimConfigFile(const std::string& yaml_content)
+{
+    creationFolder_ = inputFolder_;
+    createYamlFile("optim-config.yml", yaml_content);
+}
+
+void CreateInputFileFixture::createYamlFile(const std::string& filename,
+                                            const std::string& yaml_content)
+{
+    std::filesystem::path yamlPath = creationFolder_ / filename;
+    std::ofstream outStream;
+    outStream.open(yamlPath, std::ofstream::trunc | std::ofstream::out);
+    outStream << yaml_content;
+    outStream.flush();
+    outStream.close();
+}
```

Comment:
The destructor of `std::ofstream` closes the file this for you
```suggestion
}
```

Reply:
Done